### DISCUSSION TITLE
How to turn off caching

### DISCUSSION THREAD

**Discussion by NiJeTi** [Discord-Net-Labs/Discord.Net-Labs#454] [Q&A]
Hello!

Is there any way to turn off or at least reset caching in slash commands?

I have a slash command that sets user limit to voice channel. It takes voice channel caller currently in, takes it's number of users and sets the limit. But because of caching the actual behavior is unpredictable. It can lock another channel where user was an hour ago or lock it to wrong number of users.

Thank you!

**Comment by bman46** [comment]
You can just delete and re-add all of the slash commands. The caching is internal to Discord and cannot be disabled. 

The following code worked for me:
```
// Load all registered commands:
var commands = await _client.Rest.GetGlobalApplicationCommands();
// Delete all commands:
foreach(var command in commands)
{
    await command.DeleteAsync();
}
// Re-register commands:
await _interact.RegisterCommandsGloballyAsync(true);
```
[You can view this in context in my bot's code.](https://github.com/bman46/InstagramEmbedDiscordBot/blob/d6072702c289dd6672a7b44d34073f22f259adce/Instagram%20Reels%20Bot/Services/CommandHandler.cs#L217)

**Reply by NiJeTi** [reply]
If I convert all my commands from global to per-guild local, will it fix my problem? Unfortunately, there is no information about caching in Discord docs

**Reply by bman46** [reply]
Don't convert them to guild commands. That won't scale well and could cause more issues.

**Reply by NiJeTi** [reply]
My bot is not public and only in use on 1 server, so I think scaling problems can be ignored for the moment. Will conversion fix the caching issue?

**Reply by NiJeTi** [reply]
I've tested both yours and mine solution and they did not resolve my problem

**Reply by bman46** [reply]
I can confirm that my method works as I used it myself. Did you wait an hour for the changes to process? It’s not instant.

**Comment by csmir** [comment]
You don't need to do any of this. Discord has an endpoint for application commands called `BulkOverwrite`. if you call `client.Rest.BulkOverwriteApplicationCommands(range)`, ALL old commands will be replaced with the new set you're uploading.
